### ISSUE TITLE
Inline footnotes

### ISSUE THREAD

**Issue by alerque** [simoncozens/sile#372]
The entire insertions system seems to currently assume a block level insertion layout where each incoming item is in a separate vbox. I'm going to need to a feature to typeset footnotes inline with previous footnotes.

**Comment by simoncozens** [comment]
I have something vaguely started on this, but I have realised that it's poking at the wrong insertion box. Instead of manipulating the insertion box for the page, it should manipulate the first insertion box on the page.

**Comment by jtauber** [comment]
Any progress on this? 

I'm doing a variant of the footnote package for critical apparatus but I want the insertions to be inline.

I can probably fund work on this.

**Comment by alerque** [comment]
@jtauber I have a couple projects that do this. I forget how I implemented it but I'll dig it up for you.

**Comment by simoncozens** [comment]
That would be helpful - I am finishing off a major project and wouldn't be able to spend much time on this until June or so.

**Comment by alerque** [comment]
Hmm, neither of the two projects I was thinking of are actually doing quite what you want.

I have two kinds of "inline" footnotes working.

1. Actually inline to the text. This is an alternative to putting footnotes in footnotes at all and basically just substitutes the footnote marker for a superscript with the actual footnote content. I've found this easier for some proofreading situations, and the same document can later be set with footnotes below. This looks something like this:

    ![image](https://user-images.githubusercontent.com/173595/79246164-c3dad180-7e81-11ea-9088-25287feec21e.png)

2. Keep the footnotes together in a block, but rather than putting the block in a footnote frame put it inline to the actual document. I'm using this for example with catechisms where excessive amounts of footnotes are better kept grouped together under each question rather than flooding the page with a footnote frame. This looks something like this:

    ![image](https://user-images.githubusercontent.com/173595/79245033-3b0f6600-7e80-11ea-93c2-1a75ca96dd96.png)

What this issue was opened for and I think you probably want is more like the latter of those examples where a number of footnotes are typeset run together in a single paragraph (rather than with line breaks between them) but are typeset in an insertion frame at the bottom of the page that takes away from the content frame.

This will require quite a bit more development than what I have because there is more math and node queue shenanigans to deal with, but I think it can be done. I'll try to help make it happen if that's what you are after.

**Comment by jtauber** [comment]
Here is an example of what I'm trying to achieve:

<img width="642" alt="Screen Shot 2020-04-15 at 12 30 06 AM" src="https://user-images.githubusercontent.com/24426/79249778-58f4bf00-7eb0-11ea-9e39-65d461fb71b1.png">

and this is what I'm currently able to achieve:

<img width="705" alt="Screen Shot 2020-04-15 at 12 30 20 AM" src="https://user-images.githubusercontent.com/24426/79249794-5d20dc80-7eb0-11ea-8a17-527c9282ea89.png">

**Comment by jtauber** [comment]
My package, copied from the footnotes package has:

```
  local material = SILE.call("vbox", {}, function ()
    SILE.call("appcrit:font", {}, function ()
      SILE.call("noindent")
      SILE.process(content)
    end)
  end)
```
and I was hoping it would be as simple as not calling `vbox` but alas that's not the case.

**Comment by alerque** [comment]
Yes, the function that calculates the height of the footnote frame relies on having multiple discrete points to fall back on if things don't fit. That in turn provides feedback to the page break algorithm. If you take away the wrappers from each footnote it can't decide whether _that_ footnote still fits on the page when it encounters a new footnote & marker. I'll have to rig up a different way of recalculating the size of insertions.

Are you able to point me to / grant me access to the data and source you are using for the actual project? I can certainly rig up some dummy stuff to play with but it's way more fun to work on real data.

**Comment by jtauber** [comment]
I'll push to a repo and grant you access (probably tomorrow). Thanks!

**Comment by ousia** [comment]
Any news for line numbering in main text and line footnotes, as required for critical editions?

Many thanks for your help.

**Comment by alerque** [comment]
@ousia I *think* I know what you're talking about and I also *think* I worked up a sample for @jtauber that implemented this, but perhaps you can post an example of what you mean (in another issue preferably since this issue isn't really related) and if it's a match I'll work up an example that isn't in James' private repo).
